carball/analysis/hit_detection/hit_analysis.py

```diff
@@ -199,6 +199,13 @@ def find_hit_stats(data_frame: pd.DataFrame, player_map: Dict[str, Player],
             simulation_time = time.time()
             total_simulation_time += simulation_time - stat_time
 
+            try:
+                if player_map[previous_saltie_hit.player_id.id].is_orange != player_map[saltie_hit.player_id.id].is_orange:
+                    if previous_saltie_hit.shot and not previous_saltie_hit.goal:
+                        saltie_hit.save = True
```

Comment:
Can you add a unit test for this?
We should have a replay that already looks at saves